common/lib/database.py

```diff
@@ -63,25 +84,37 @@ def query(self, query, replacements=None, cursor=None):
 		if not cursor:
 			cursor = self.get_cursor()
 
-		self.log.debug("Executing query %s" % self.cursor.mogrify(query, replacements))
-
-		return cursor.execute(query, replacements)
+		self.log.debug("Executing query %s" % cursor.mogrify(query, replacements))
+		try:
+			cursor.execute(query, replacements)
+		except (psycopg2.InterfaceError, psycopg2.OperationalError) as e:
```

Comment:
There is a log on line 59 for a reconnect, but I would probably add at least a warning-level log here that a query failed due to a connection error, that includes (part of) the original exception (perhaps in place of the line 59 log). That would make it easier to know what exactly failed, and whether it's the fault of the database or 4CAT, etc. If we only log a warning in `reconnect` we lose info about the query itself.

Reply:
Makes sense to me. Refactored to warn prior to calling reconnect.